components/brave_adblock_ui/adblock_internals/components/app.tsx

```diff
@@ -15,37 +15,91 @@ class AppState {
   memory: { [key: string]: string } = {}
 }
 
+const EngineInfo = (props: { engine: EngineDebugInfo; caption: string }) => {
+  return (
+    <div>
+      <h2>{props.caption}</h2>
+      <div>
+        Flatbuffer size:{' '}
+        {(props.engine.flatbuffer_size / 1024 / 1024).toFixed(2)} MB
+      </div>
+      <div>Compiled regexes: {props.engine.compiled_regex_count}</div>
+    </div>
+  )
+}
+
 export class App extends React.Component<{}, AppState> {
-  constructor (props: {}) {
+  constructor(props: {}) {
     super(props)
     this.state = new AppState()
     this.getDebugInfo()
-    setInterval(() => { this.getDebugInfo() }, 2000)
+    setInterval(() => {
+      this.getDebugInfo()
+    }, 2000)
   }
 
-  getDebugInfo () {
+  getDebugInfo() {
     return sendWithPromise('brave_adblock_internals.getDebugInfo').then(
-      this.onGetDebugInfo.bind(this))
+      this.onGetDebugInfo.bind(this),
+    )
   }
 
-  onGetDebugInfo (state: AppState) {
+  onGetDebugInfo(state: AppState) {
     saveRegexTexts(state.default_engine.regex_data)
     saveRegexTexts(state.additional_engine.regex_data)
     this.setState(state)
   }
 
-  discardAll () {
+  discardAll() {
     discardRegexs(this.state.default_engine.regex_data)
     discardRegexs(this.state.additional_engine.regex_data)
   }
 
-  render () {
+  render() {
     return (
       <div>
-        <MemoryInfo key="memory" caption="Browser process memory" memory={this.state.memory} />
-        <input type="button" value="Discard All Regex" onClick={() => { this.discardAll() }} />
-        <Engine key="default_engine" caption="Default engine" info={this.state.default_engine} />
-        <Engine key="additional_engine" caption="Additional engine" info={this.state.additional_engine} />
+        <div>
+          <EngineInfo
+            engine={this.state.default_engine}
+            caption='Default engine'
+          />
+          <EngineInfo
+            engine={this.state.additional_engine}
+            caption='Additional engine'
+          />
+          <MemoryInfo
+            key='memory'
+            caption='Browser process general memory'
+            memory={this.state.memory}
+          />
+          <input
+            type='button'
+            value='Show Regexes'
+            onClick={() => {
+              window.location.href = './?regex_debug'
+            }}
+          />
+          <input
+            type='button'
+            value='Discard All Regexes'
+            onClick={() => {
+              this.discardAll()
+            }}
+          />
+        </div>
+
+        {window.location.href.includes('regex_debug') && (
```

Comment:
just fwiw the component won't necessarily rerender when this changes - though I guess we're triggering a page refresh anyway so 🤷